### PR TITLE
Allow key paths to start with "not" in queries

### LINKED ISSUE #445: field name cannot start with 'not',  such as 'notic'.
there are two query string:

1. filtered( '$0==notic',notic)
2. filtered( 'notic==$0',notic)

first is correct, second is error. 
I think that  second string is equal with ' not  ic==$0'.

can you understand?

### PR DESCRIPTION
**Pull request by appden**
Fixes #445

### CODE REVIEW

File: src/object-store/src/parser/parser.cpp [hunk 1]
```diff
@@ -24,6 +24,8 @@
 #include <pegtl/analyze.hh>
 #include <pegtl/trace.hh>
 
+#define string_operator_t(s) seq< pegtl_istring_t(s), not_at< identifier_other > >
```
Comment (alazier) [inline src/object-store/src/parser/parser.cpp]: comment here would be nice.

Comment (alazier) [inline src/object-store/src/parser/parser.cpp]: naming is ambiguous as well

**Comment by alazier** [comment]
Need to fix naming/comments for our macro. Actual logic looks good.

File: src/object-store/tests/parser.cpp [hunk 2]
```diff
@@ -66,6 +66,9 @@ static std::vector<std::string> valid_queries = {
     "!(0=0)",
     "! (0=0)",
     "NOT0=0",   // keypath NOT0
+    "NOT0.a=0", // keypath NOT0
```
Comment (alazier) [inline src/object-store/tests/parser.cpp]: maybe add a similar test for AND/OR

File: src/object-store/src/parser/parser.cpp [hunk 3]
```diff
@@ -93,11 +95,11 @@ struct group_pred : if_must< one< '(' >, pad< pred, blank >, one< ')' > > {};
...
 struct false_pred : pegtl_istring_t("falsepredicate") {};
 
-struct not_pre : seq< sor< one< '!' >, pegtl_istring_t("not") > > {};
+struct not_pre : seq< sor< one< '!' >, string_operator_t("not") > > {};
 struct atom_pred : seq< opt< not_pre >, pad< sor< group_pred, true_pred, false_pred, comparison_pred >, blank > > {};
 
-struct and_op : pad< sor< two< '&' >, pegtl_istring_t("and") >, blank > {};
-struct or_op : pad< sor< two< '|' >, pegtl_istring_t("or") >, blank > {};
+struct and_op : pad< sor< two< '&' >, string_operator_t("and") >, blank > {};
+struct or_op : pad< sor< two< '|' >, string_operator_t("or") >, blank > {};
```
Comment (bdash) [inline src/object-store/src/parser/parser.cpp]: Is a similar change needed for `TRUE`, `FALSE`, `NULL`, `TRUEPREDICATE`, `FALSEPREDICATE`, `CONTAINS`, `BEGINSWITH`, and `ENDSWITH`?

Comment (appden) [inline src/object-store/src/parser/parser.cpp]: @bdash I opened PR #456 that I believe fixes the rest of the issues you brought up.

**Pull request merged**